README.md

```diff
@@ -1 +1,91 @@
-Envoy v2 gRPC APIs
+# Envoy v2 gRPC APIs
+
+## Goals
+
+This repository contains the draft v2 gRPC
+[Envoy](https://github.com/lyft/envoy/) APIs. Envoy today has a number of JSON
+REST APIs through which it may discover and have updated its runtime
+configuration from some management server. These are:
+
+* [Cluster Discovery Service (CDS)](https://lyft.github.io/envoy/docs/configuration/cluster_manager/cds.html)
+* [Rate Limit Service (RLS)](https://lyft.github.io/envoy/docs/configuration/overview/rate_limit.html)
+* [Route Discovery Service (RDS)](https://lyft.github.io/envoy/docs/configuration/http_conn_man/rds.html)
+* [Service Discovery Service (SDS)](https://lyft.github.io/envoy/docs/configuration/cluster_manager/sds_api.html)
+
+Version 2 of the Envoy API will evolve existing APIs and introduce new APIs to:
+
+* Allow for more advanced load balancing through load and resource utilization reporting to management servers.
+* Improve N^2 health check scalability issues by optionally offloading health checking to other Envoy instances.
+* Support Envoy deployment in edge, sidecar and middle proxy deployment models via changes to the listener model and CDS/SDS APIs.
+* Allow streaming updates from the management server on change, instead of polling APIs from Envoy. gRPC APIs will be supported
+  alongside JSON REST APIs to provide for this.
+* Ensure all Envoy runtime configuration is dynamically discoverable via API
+  calls, including listener configuration, certificates and runtime settings, which are today sourced from the filesystem.
+* Revisit and where appropriate cleanup any v1 technical debt.
+
+## Status
+
+Draft work-in-progress. Input is welcome via issue filing. Small, localized PRs
+are also welcome, but any major changes or suggestions should be coordinated in
+a tracking issue with the authors.
+
+## Principles
+
+* [Proto3](https://developers.google.com/protocol-buffers/docs/proto3) will be
+  used to specify the canonical API. This will provide directly the gRPC API and
+  via gRPC-JSON transcoding the JSON REST API.
+
+* xDS APIs should support eventual consistency. For example, if HDS references a
+  host that has not yet been supplied by EDS, it should be silently ignored.
```

Comment:
I'm wondering why HDS and EDS can't be eventually consistent. What happens if HDS serves up its response slightly before EDS? Should the configuration be rejected, or accepted pending an EDS response which includes the HDS endpoints?